docker-compose.yml

```diff
@@ -149,10 +150,39 @@ services:
       - "bundle:/usr/local/bundle"
       - "tmp-test:/home/dev/openproject/tmp"
 
+  ci:
+    build:
+      context: ./docker/ci/
+      dockerfile: Dockerfile
+    command: setup run-units
+    hostname: ci
+    environment:
+      RAILS_ENV: test
+      OPENPROJECT_RAILS__RELATIVE__URL__ROOT: "${OPENPROJECT_RAILS__RELATIVE__URL__ROOT:-}"
+      SELENIUM_GRID_URL: http://selenium-hub:4444/wd/hub
+      CAPYBARA_DYNAMIC_HOSTNAME: 0
+      CAPYBARA_APP_HOSTNAME: ci
+      OPENPROJECT_DISABLE_DEV_ASSET_PROXY: 1
+      OPENPROJECT_TESTING_NO_HEADLESS: "true"
+      JOBS: "${CI_JOBS:-4}"
+    tmpfs:
+      - "/tmp"
+    volumes:
+      - ".:/home/dev/openproject"
+      - "${CI_CACHE_PATH:-/tmp/}op-bundle:/usr/local/bundle"
+      - "${CI_CACHE_PATH:-/tmp/}op-node:/home/dev/openproject/frontend/node_modules"
+    networks:
+      - testing
+    depends_on:
+      - selenium-hub
+
+  # https://vitobotta.com/2019/09/04/rails-parallel-system-tests-selenium-docker/
   selenium-hub:
     image: selenium/hub:latest
     container_name: selenium-hub
     hostname: selenium-hub
+    environment:
+      GRID_MAX_SESSION: 16
```

Comment:
We could perhaps even set this in the env, allowing you to change it locally / on ci

Reply:
Yes you're right it should use the same `CI_JOBS` var used to specify the number of parallel jobs.